nexus/reconfigurator/blippy/src/blippy.rs

```diff
@@ -0,0 +1,398 @@
+                     zpool {durable_zpool} but its root dataset on \
+                     zpool {transient_zpool}",
+                    zone.zone_type.kind(),
+                    zone.id,
+                )
+            }
+            Kind::SledMissingDatasets { why } => {
+                write!(f, "missing entry in blueprint_datasets ({why})")
+            }
+            Kind::SledMissingDisks { why } => {
+                write!(f, "missing entry in blueprint_disks ({why})")
+            }
+            Kind::OrphanedDataset { dataset } => {
+                let parent = match dataset.kind {
+                    DatasetKind::Cockroach
+                    | DatasetKind::Crucible
+                    | DatasetKind::Clickhouse
+                    | DatasetKind::ClickhouseKeeper
+                    | DatasetKind::ClickhouseServer
+                    | DatasetKind::ExternalDns
+                    | DatasetKind::InternalDns
+                    | DatasetKind::TransientZone { .. } => "zone",
+                    DatasetKind::TransientZoneRoot
+                    | DatasetKind::Debug
+                    | DatasetKind::Update => "disk",
+                };
+                write!(
+                    f,
+                    "in-service dataset ({:?} {}) with no associated {parent}",
+                    dataset.kind, dataset.id
+                )
+            }
+            Kind::DatasetOnNonexistentZpool { dataset } => {
+                write!(
+                    f,
+                    "in-service dataset ({:?} {}) on non-existent zpool {}",
+                    dataset.kind, dataset.id, dataset.pool
+                )
+            }
+        }
+    }
+}
+
+impl Note<'_> {
+    pub fn display(&self, sort_key: BlippyReportSortKey) -> NoteDisplay<'_> {
+        NoteDisplay { note: self, sort_key }
+    }
+}
+
+#[derive(Debug)]
+pub struct NoteDisplay<'a> {
+    note: &'a Note<'a>,
+    sort_key: BlippyReportSortKey,
+}
+
+impl fmt::Display for NoteDisplay<'_> {
+    fn fmt(&self, f: &mut fmt::Formatter<'_>) -> fmt::Result {
+        match self.sort_key {
+            BlippyReportSortKey::Component => {
+                write!(
+                    f,
+                    "{}: {} note: {}",
+                    self.note.component, self.note.severity, self.note.kind
+                )
+            }
+            BlippyReportSortKey::Severity => {
+                write!(
+                    f,
+                    "{} note: {}: {}",
+                    self.note.severity, self.note.component, self.note.kind
+                )
+            }
+        }
+    }
+}
+
+#[derive(Debug)]
+pub struct Blippy<'a> {
+    blueprint: &'a Blueprint,
+    notes: Vec<Note<'a>>,
+}
+
+impl<'a> Blippy<'a> {
+    pub fn new(blueprint: &'a Blueprint) -> Self {
+        let mut slf = Self { blueprint, notes: Vec::new() };
+        checks::perform_all_blueprint_only_checks(&mut slf);
+        slf
+    }
```

Comment:
My intent here is to add at least one more method here that would let us perform additional checks that need the `PlanningInput` (or at least the `Policy`), but make that optional so it can still be used to check just a standalone blueprint. So something like:

```rust
let just_blueprint = Blippy::new(blueprint).into_report();                      // today
let full_context   = Blippy::new(blueprint).with_context(policy).into_report(); // future
```